test/functional/wallet-encryption.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+# Copyright (c) 2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test Wallet encryption"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_jsonrpc,
+)
+import os
+import shutil
+import time
+
+class WalletEncryptionTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def run_test (self):
+        tmpdir = self.options.tmpdir
+        passphrase = "WalletPassphrase"
+        passphrase2 = "SecondWalletPassphrase"
+        
+        # Make sure the wallet isn't encrypted first
+        address = self.nodes[0].getnewaddress()
+        privkey = self.nodes[0].dumpprivkey(address)
+        
+        # Encrypt the wallet
+        self.nodes[0].encryptwallet(passphrase)
+        self.stop_node(0)
```

Comment:
This call to `stop_node()` is not needed (and may be dangerous). the encryptwallet RPC causes bitcoind to stop. Really what you need to do here is wait for the bitcoind process to terminate rather than call `stop_node()`. See https://github.com/bitcoin/bitcoin/blob/9c248e39f2807a7f89e555e99cc55cb2de1ad335/test/functional/fundrawtransaction.py#L455 for example.

`stop_node()` will assert if the bitcoind process no longer exists, so your current code is a bit racy. If the call to encryptwallet causes the node to stop before stop_node() is called, then the test will assert and fail.